File: lagan/src/main/kotlin/com/valet/lagan/model/Event.kt
package com.valet.lagan.model

import java.util.*

data class Event(
    private val id: UUID,
    private val usernameCreator: String,
    private val name: String,
    private val address: String,
    private val photoLink: String,
    private val peopleMaxSize: UInt,
    private val peopleMinSize: UInt,
    private val peopleSize: UInt,
    private val peopleFriendSize: UInt,
    private val startTime: Date,
    private val endTime: Date,
    private val accessType: EventType
)

enum class EventType {
    ALL_USER, FRIEND_ONLY, BY_INVITATION, BY_INVITATION_OR_REQUEST
}

File: internal/model/event.go
package model

import "time"

type Event struct {
	Id            string
	UserId        string
	Name          string
	Address       string
	PhotoLink     string
	PeopleMaxSize int
	PeopleSize    int
	PeopleMinSize int
	FriendSize    int
	StartTime     time.Time
	EndTime       time.Time
	AccessType    EventAccessType
}

type EventAccessType string

const (
	AllUsers          = EventAccessType("all_users")
	FriendlyOnly      = EventAccessType("friendly_only")
	ByInvite          = EventAccessType("by_invite")
	ByInviteOrRequest = EventAccessType("by_invite_or_request")
)
